src/libmongoc/src/mongoc/mongoc-client.c

```diff
@@ -299,6 +299,25 @@ typedef bool (*mongoc_rr_callback_t) (const char *service,
                                       mongoc_rr_data_t *rr_data,
                                       bson_error_t *error);
 
+static const char *
+_mongoc_hstrerror (int code)
+{
+   switch (code) {
+   case HOST_NOT_FOUND:
+      return "The specified host is unknown.";
+   case NO_ADDRESS:
+   case NO_DATA:
+      return "The requested name is valid but does not have an IP address.";
+   case NO_RECOVERY:
+      return "A nonrecoverable name server error occurred.";
+   case TRY_AGAIN:
+      return "A temporary error occurred on an authoritative name server. Try "
+             "again later.";
+   default:
+      return "An unknown error occurred.";
```

Comment:
In practice, would we ever hit this case? If so, I wonder if it'd be helpful to somehow expose the integer value in the error message relayed up the chain. I realize we can't print the value of `code` here since it'd require allocating a string, but perhaps we can do this in the calling context.

If it's unlikely we'd hit this case, I'm fine leaving it as-is.

Reply:
I was thinking of it but avoided it because it would require allocating memory. Either way, even if we hit this scenario, it would be difficult for us to do anything as the value given would not make any sense to us (since it isn't documented). Even with the code, we'd most likely have to resort to tracing or other means to find out what's going on so I opted for this message.